### ISSUE TITLE
v1.6.1 segfault with abi generation

### ISSUE THREAD

**Issue by johndebord** [EOSIO/eosio.cdt#511]
v1.6.1 of `eosio-cpp` currently has a bug which causes a seg fault, but only with abi generation and under certain circumstances.

**Comment by cryptosig** [comment]
I believe I'm running into this error as well

**Comment by johndebord** [comment]
@cryptosig This issue has been resolved. The problem we encountered was that the tool `eosio-abigen` segfaults when declaring a `std::vector` of itself within its own data structure.

<hr>

For example:
```c++
#include <vector>

#include <eosio/contract.hpp>
#include <eosio/multi_index.hpp>

using namespace eosio;

struct [[eosio::contract]] main : eosio::contract {
   using contract::contract;

   struct [[eosio::table]] structure {
      std::vector<structure> vec;
   };

   [[eosio::action]] void act()
   { }
};
```
Compiles, but segaults during `.abi` generation.

<hr>

```c++
#include <vector>

#include <eosio/contract.hpp>
#include <eosio/multi_index.hpp>

using namespace eosio;

struct [[eosio::contract]] con : eosio::contract {
   using contract::contract;

   struct [[eosio::table]] structure {
      std::vector<int> vec;
   };

   [[eosio::action]] void act()
   { }
};
```
Compiles and does not segault during `.abi` generation.

**Comment by znebby** [comment]
@johndebord How is this issue solved? 1.6.2 still SegFaults when a table references itself in a vector. Can we re-open this issue?

**Comment by jeffreyssmith2nd** [comment]
@znebby See issue #527, there's an ongoing discussion on that issue about table references in a vector.
